datafusion/physical-expr/src/expressions/binary.rs

```diff
@@ -4120,4 +4187,143 @@ mod tests {
             .contains("Overflow happened on: 2147483647 * 2"));
         Ok(())
     }
+
+    macro_rules! test_regex_match_scalar {
+        (
+            $A_ARRAY:ident, 
+            $A_TYPE:expr, 
+            $A_VEC:expr, 
+            $B_SCALAR:expr, 
+            $OP:expr, 
+            $C_ARRAY:ident, 
+            $C_TYPE:expr, 
+            $VEC:expr,
+        ) => {{
+            let schema = Schema::new(vec![
+                Field::new("a", $A_TYPE, false),
+            ]);
+            let a = $A_ARRAY::from($A_VEC);
+
+            let left = col("a", &schema).unwrap();
+            let right = lit($B_SCALAR);
+
+            // verify that we can construct the expression
+            let expression = binary(left, $OP, right, &schema).unwrap();
```

Comment:
I think we should do an extra check: the regex is compiled inside the expression.